cmd/weaver/main.go

```diff
@@ -74,11 +74,25 @@ func main() {
 	switch flag.Arg(0) {
 	case "generate":
 		generateFlags := flag.NewFlagSet("generate", flag.ExitOnError)
+		tags := generateFlags.String("tags", "not specified", "Optional tags for the generate command")
```

Comment:
I am not quite sure why we need "not specified". Can't we just say that if empty, we add no tags except for our internal weaver tag, and if non-empty, we add the specified tags + our internal weaver tag. 

If we really need to detect whether or not tags were specified on the command line, a less messy alternative to "not specified" would be to call flags.Visit() to see if it reports "tags" as one of the flags set on the command line.

Reply:
Good point. I think I implemented this before understanding that I need to define tags as a flag on generateFlags instead of defining a new flag set.